lib/utils/qemu.coffee

```diff
@@ -0,0 +1,88 @@
+Promise = require('bluebird')
+
+exports.QEMU_VERSION = QEMU_VERSION = 'v2.5.50-resin-execve'
+exports.QEMU_BIN_NAME = QEMU_BIN_NAME = 'qemu-execve'
+
+exports.installQemuIfNeeded = Promise.method (emulated, logger) ->
+	return false if not (emulated and platformNeedsQemu())
+
+	hasQemu()
+	.then (present) ->
+		if !present
+			logger.logInfo('Installing qemu for ARM emulation...')
+			installQemu()
+	.return(true)
+
+exports.qemuPathInContext = (context) ->
+	path = require('path')
+	binDir = path.join(context, '.resin')
+	binPath = path.join(binDir, QEMU_BIN_NAME)
+	path.relative(context, binPath)
+
+exports.copyQemu = (context) ->
+	path = require('path')
+	fs = require('mz/fs')
+	# Create a hidden directory in the build context, containing qemu
+	binDir = path.join(context, '.resin')
+	binPath = path.join(binDir, QEMU_BIN_NAME)
+
+	fs.access(binDir)
```

Comment:
Edge case: is binDir is a file the following will fail

Reply:
Yeah, this file was merely code extracted from docker.coffee that I felt deserved its own place. In any case, I've updated it to try to create the folder unconditionally.